static/sass/_pattern_blog-card.scss

```diff
@@ -82,8 +82,21 @@
     }
   }
 
+  // a fallback to catch any instances where a post.group exists,
+  // but we haven't yet defined a BEM variant to match
+  [class*="blog-p-card__header"] {
+    @extend %post-card-header;
+    border-top: 3px solid #666; //sass-lint:disable-line no-color-literals
+  }
+
   .blog-p-card__header {
     @extend %post-card-header;
+    border-top: 3px solid #666; //sass-lint:disable-line no-color-literals
```

Comment:
Since you've already got the fallback you can remove the `@extend`s for each variant - not gonna result in any less css but it's kind of redundant. Also, all you need is `border-color` for each variant because the width and style is already defined. Also also, Vanilla has `#666` as one of its defaults (`$color-mid-dark`) so you can just use that: https://docs.vanillaframework.io/en/settings/color-settings

So it would look like:
``` scss
[class*="blog-p-card__header"] {
  @extend %post-card-header;
  border-top: 3px solid $color-mid-dark;
}

.blog-p-card__header {
  &--category1 {
    border-color: $color-category-1;
  }

  &--category2 {
    border-color: $color-category-2;
  }
...
}
```